src/app/map/draw-polygon/draw-polygon.directive.ts

```diff
@@ -64,38 +67,64 @@ export class DrawPolygonDirective implements OnInit, AfterViewInit {
     }
   }
 
-
   generatePointGrid(e) {
-    console.log('generate called');
     var data = this.drawControl.getAll();
+    let promises = [];
     if (data.features.length > 0) {
-      let points = turf.pointGrid(
-        turf.bbox(
-          data.features[0]),
-        30,
-        {
-          units: 'meters',
-          mask: data.features[0]
-        });
       let map = this.parent.mapInstance;
-      map.addSource('points',
-        {
-          "type": "geojson",
-          "data": points
-        })
-      map.addLayer({
-        "id": "points",
-        "type": "symbol",
-        "source": "points",
-        "layout": {
-          "icon-image": "custom-marker",
-          "text-field": "{title}",
-          "text-font": ["Open Sans Semibold", "Arial Unicode MS Bold"],
-          "text-offset": [0, 0.6],
-          "text-anchor": "top"
+      let promiseToGetMarkers = new Promise((res, rej) => {
+        let points = turf.pointGrid(
+          turf.bbox(
+            data.features[0]),
+            30,
+            {
+            units: 'meters',
+            mask: data.features[0]
+            }
+        );
+        let confirmedPoints = [];
+        console.log(points);
+        points.features.forEach((item, index, object) => {
+          const lat = item.geometry.coordinates[1];
+          const lon = item.geometry.coordinates[0];
+          const promise = this.http.get(`https://api.onwater.io/api/v1/results/${lat},${lon}?access_token=Z6j-_YLQkwfMoSPsG_c4`).toPromise().then(data => {
```

Comment:
Deze API heeft een max aantal requests per minuut op het onbetaalde platform ( 15 dacht ik te lezen ). Met een grote selectie overschrijden we deze max aantal requests met gemak, we moeten dus een alternatief bedenken om dit op te lossen ipv in een forEach. Denk bijvoorbeeld aan een complete collectie ipv een enkel coordinaat als request.